server/src/main/java/io/crate/expression/reference/doc/lucene/SourceParser.java

```diff
@@ -162,7 +162,8 @@ private static Map<String, Object> parseObject(XContentParser parser,
                                                    Set<String> droppedColumns,
                                                    StringBuilder colPath,
                                                    boolean includeUnknown) throws IOException {
-        if (requiredColumns == null || requiredColumns.isEmpty()) {
+        if ((requiredColumns == null || requiredColumns.isEmpty()) && droppedColumns.isEmpty()) {
+            // If there are dropped columns, we cannot return source without parsing, we need to hide dropped column.
```

Comment:
this was the problem for approach 1 - we don't register refs via `SourceLookup.registerRef` in `RowCollectExpression` thus we fall to this "return as is" behavior.

I tried to tackle this by forcing parsing with extra `droppedColumns.isEmpty()` check but then some code below also doesn't work with empty requiredColumns. 

I think just using _doc is a cleaner approach - as we essentially doing smth similar to _doc collection, just don't need to inject partition values.

Reply:
_doc also wouldn't work, since we also don't register requiredColumns.
In general, if `select _raw` does parsing, ALL table columns must be included into `requredColumns`.

Will get back to the first approach and explicitly add all table columns.